game-core/src/main/java/games/strategy/triplea/ui/logic/RouteCalculator.java

```diff
@@ -25,14 +31,6 @@
   private final int mapWidth;
   private final int mapHeight;
 
-  public RouteCalculator(final boolean isInfiniteX, final boolean isInfiniteY, final int mapWidth,
```

Comment:
In what order does `@AllArgsConstructor` list the parameters?  I'm assuming it's in the same order that the fields are declared.  If so, then the Lombok version will reverse `isInfiniteX` and `isInfiniteY` from the current version, and all call sites will need to be updated.

Since the constructor is private, my question above may be moot because there are no manual calls to the constructor and all calls will be via the generated Lombok builder.  I guess it will map the builder state to the correct arguments since it knows in what order it generated the constructor parameters.

Please ignore; just thinking out loud... :smile:

Reply:
> In what order does @AllArgsConstructor list the parameters? I'm assuming it's in the same order that the fields are declared

That is correct. This is a pretty good example of the API cleanups we can get, quick parameter objects can be created and then passed instead of having a many arg method.

Reply:
I added another example that uses a similar pattern to help clean up a `boolean` parameter. The builder gives us a label and takes away ordering. This is can often help with type safety, like here where there are two parameters of the same type side by side.

944bbd6